src/js/Parser.js

```diff
@@ -0,0 +1,92 @@
+import { Station } from './Station';
+import { SubwayLine } from './SubwayLine';
+import { Subway } from './Subway';
+
+export function parseToSubway(data) {
+    let currentData = data;
+
+    currentData = handleToArray(currentData);
+    currentData = handleToSubway(currentData);
+
+
+    return createSubway(currentData);
+} 
+
+function handleToArray(data) {
+    let array = data;
+    array = array.split('\n');
+    array = Array.from(array);
+
+    array = array.map(name => {
+        return name.trim();
+    });
+
+    return array;
+}
+
+function handleToSubway(data) {
+    let stationNames = [], names = [], lineNames = [];
+    let lineName = null;
+
+    data.forEach(name => {
+        if (!name || data.lastIndexOf(name) === data.length - 1) {
+            if (data.lastIndexOf(name) === data.length - 1) {
+                names.push(name);
+            }
+            lineName = names.shift();
+            stationNames.push(names);
+            lineNames.push(lineName);
+            names = [];
+        } else {
+        names.push(name);
+        }
+    });
+
+    return { lineNames: lineNames, stationNames: stationNames };
+}
+
+function createSubway(data) {
+    let stationNames = data.stationNames.concat(),
```

Comment:
Please do not use single let when you declare variables. Use separate let for each declaration